### PR TITLE
Remove recently added wpcom.undocumented() methods

### PR DESCRIPTION
**Pull request by jsnajdr**
We'd like to remove all `wpcom.undocumented()` methods and to replace them with direct `wpcom.req.get` or `wpcom.req.post` calls. The reason is to reduce bundle size and remove the rather large `wpcom-undocumented` module.

In 2021 several `wpcom.undocumented()` methods were _added_ instead, going against our strategy. This PR migrates them:
- `restoreAtomicPlanSoftware`: a simple POST used at only one place (@brandonpayton)
- `getAtomicSiteLogs`: a POST request (btw why isn't it a GET when it only reads logs?) used at one place (@delputnam)
- keyring connections for P2 and classic sites: 4 methods all used in one actions module (@unDemian)
- Titan: several GETs for Titan email (@daledupreez)

**How to test:**
Verify that the affected features work correctly.

### CODE REVIEW

File: client/lib/wpcom-undocumented/lib/undocumented.js [hunk 1]
```diff
@@ -620,88 +620,6 @@ Undocumented.prototype.validateDomainContactInformation = function (
...
- */
-Undocumented.prototype.getTitanOrderProvisioningURL = function ( domain, fn ) {
-	return this.wpcom.req.get(
-		{
-			path: `/emails/titan/${ encodeURIComponent( domain ) }/order-provisioning-url`,
-			apiNamespace: 'wpcom/v2',
-		},
-		fn
-	);
-};
```
Comment (stephanethomas) [inline client/lib/wpcom-undocumented/lib/undocumented.js:637]: For the records, this method was no longer used.

**Review by stephanethomas — COMMENTED** [review]
Thanks for cleaning up that code Jarda! I've checked changes for Titan, and asserted that the following still work:

* Checking mailbox availability on the `Add New Mailboxes` page
* Being redirected to the `Purchases` page from the Titan Control Panel
* Going to the Titan Control Panel from the `Manage all mailboxes` page

I didn't test auto-logging into the Titan Control Panel from the iframe though since we no longer support that feature. So for Titan this pull request is green ✅.

File: client/lib/wpcom-undocumented/lib/undocumented.js (hunk 1 above)
Comment (jsnajdr) [reply client/lib/wpcom-undocumented/lib/undocumented.js:637]: Thanks, that's a good clarification :+1:

**Review by brandonpayton — COMMENTED** [review]
The changes around `restoreAtomicPlanSoftware()` seem to be working well.

Thanks for catching this, @jsnajdr. If people should not be adding to this file, I wonder if there should be automation enforcing that, so people who don't know the general direction of these kinds of things can be notified instead of you having to notice and correct. Maybe automation is overkill for this case, but if we have other such cases where we want to reduce the size of a file to zero, maybe it would be worth it.

**Comment by jsnajdr** [comment]
> I wonder if there should be automation enforcing that

I'm not sure if it worth doing, as folks are accidentally adding new methods like twice a year, but in case we did it, I think the best solution is an ESLint rule that catches calls to `wpcom.undocumented()` where `wpcom` is the binding imported from `calypso/lib/wp`.

**Comment by unDemian** [comment]
> We'd like to remove all wpcom.undocumented() methods and to replace them with direct wpcom.req.get or wpcom.req.post calls. The reason is to reduce bundle size and remove the rather large wpcom-undocumented module.

Sorry about that, I was not aware of this effort.

> keyring connections for P2 and classic sites: 4 methods all used in one actions module (@unDemian)

Looking good, thanks for doing this.

File: client/my-sites/hosting/restore-plan-software-card/index.js [hunk 2]
```diff
@@ -1,35 +1,36 @@
...
+		wpcom.req
+			.post( {
+				path: `/sites/${ siteId }/hosting/restore-plan-software`,
+				apiNamespace: 'wpcom/v2',
+			} )
 			.then( () => {
-				showSuccessNotice(
-					translate( 'Requested restoration of plugins and themes that come with your plan.' )
+				dispatch(
+					successNotice(
```
Comment (tyxla) [inline client/my-sites/hosting/restore-plan-software-card/index.js:24]: Much clearer 👍

File: client/my-sites/hosting/web-server-logs-card/index.js [hunk 3]
```diff
@@ -130,9 +130,19 @@ const WebServerLogsCard = ( props ) => {
 		let isError = false;
 
 		do {
-			await wpcom
-				.undocumented()
-				.getAtomicSiteLogs( siteId, startTime, endTime, scrollId )
```
Comment (tyxla) [inline client/my-sites/hosting/web-server-logs-card/index.js:135]: Noting that the old function name was bringing some confusion: when one sees `get*` in the function name, they assume it performs a `GET`. However, this one performs a `POST` request and by removing the undocumented function, we also remove this confusion 👍

**Review by tyxla — APPROVED** [review]
I took some time to inspect the changes - code-wise this looks great, and all the requests are preserved. 

I'm deferring the testing part to the responsible teams, as I'm not very familiar with all of the functionality that's tested. From my understanding it tested well for all of them, so I'm giving a green light ✅

File: client/my-sites/hosting/web-server-logs-card/index.js (hunk 3 above)
Comment (jsnajdr) [reply client/my-sites/hosting/web-server-logs-card/index.js:135]: The root cause of the confusion is that the `/logs` endpoint uses `POST` (called `WP_REST_Server::CREATABLE` in WP REST lingo) although it only reads logs. It should be a `GET`. Looks like a bug in the server implementation (@delputnam) to me.

Comment (delputnam) [reply client/my-sites/hosting/web-server-logs-card/index.js:135]: > Looks like a bug in the server implementation

The WPCOM API was created this way to match the internal API that provides similar functionality. I suppose that the WPCOM endpoint doesn't necessarily have to match the internal endpoint though.  I will see if I can find someone who is in a better position to address this in the near term. If not, I can have a look at this in the future.

**Pull request merged**